cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+2. take steps to allow  `rad-role-arn` and `eks cluster : namespace: service` to trust each other using aws portal.
+3. use  `rad init` to configure radius with role-arn credential. (or use rad credential later on)
+4. use `rad env update` to configure provider associated with the specific environment (`rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region>`). 
+5. during rad deploy, radius assumes the role that is configured in the credential. 
+
+Since oidc is configured for a cluster and not specific to a role-arn, the solution can be extended to work in multi-tenancy once we have multi credentials support. 
+
+More details about each step is covered in [Detailed Design](#detailed-design)
+
+IRSA works the same way on EKS as well as non-EKS clusters.  
+
+### Alternatives considered
+
+#### EKS Pod Identity 
+
+EKS Pod Identity was introduced in 2022 as a simplified approach for applications running on EKS to retrieve credentials. It uses the new EKS pod identity instead of OIDC provider for identity. Because of this, users have convenient APIs that allow managing pod identity. 
+Ref. [AWS EKS Pod Identity](https://aws.amazon.com/blogs/containers/amazon-eks-pod-identity-a-new-way-for-applications-on-eks-to-obtain-iam-credentials/). 
+
+Since the identity provider is tied to AWS, this solution cannot be used with Radius deployed on stand alone, aks or any other kubernetes cluster. 
+
+Its good to note that since this approach uses its own service principal, the setup involves different identity providers and configuration on aws. Radius should be however able to support this feature with minimal code changes, if any.  
+
+
+### Architecture Diagram
+
+
+### Detailed Design
+
+#### Proposed Option
+
+Radius should support AWS IRSA.
+
+#### Prerequisite
+
+1. [Setup OIDC provider] (https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html) should be complete for the cluster. 
+2. `rad-role` with desired policies should be created. Its trust relation should look similar to below. This establishes a trust relation between the annotated service-accounts in the specified namespace in specified cluster with the specified OIDC provider to be established. 
+  ```
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::817312594854:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:sub": "system:serviceaccount:radius-system:ucp",
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:aud": "sts.amazonaws.com"
+                }
+            }
+        },
+    ]
+}
+  ```
```

Comment:
In this policy document, what are the things that are static and what are the things that will change from user-to-user? Can you point out where identifiers from this document need to match something in Radius?

Reply:
I will add the radius specific explanation and capture ucp + recipe rp details.